client/src/components/CreatePost.test.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import CreatePost from "./CreatePost";
+import { unmountComponentAtNode } from "react-dom";
+import { act } from "react-dom/test-utils";
+import { render, screen, fireEvent } from "@testing-library/react";
+import newImage from "../assets/background.jpg";
+
+let container: HTMLElement | null = null;
+beforeEach(() => {
+  // setup a DOM element as a render target
+  container = document.createElement("div");
+  document.body.appendChild(container);
+});
+
+afterEach(() => {
+  // cleanup on exiting
+  if (container != null) {
+    unmountComponentAtNode(container);
+    container.remove();
+    container = null;
+  }
+});
+
+describe("verifying launch of create post component", () => {
+  it("should render out the create post form", () => {
+    act(() => {
```

Comment:
Instead of repeating this for each test we can put it into beforeEach, and in afterEach you can call cleanup() to destroy it